web/packages/teleport/src/Console/DocumentDb/ConnectDialog.tsx

```diff
@@ -0,0 +1,209 @@
+      if (response.agents.length !== 1 || response.agents[0].kind !== 'db') {
+        throw new Error('Unable to retrieve database information.');
+      }
+
+      return response.agents[0];
+    }, [props.clusterId, ctx.resourceService, props.serviceName])
+  );
+
+  useEffect(() => {
+    void getDatabase();
+  }, [getDatabase]);
+
+  return (
+    <Dialog
+      dialogCss={dialogCss}
+      disableEscapeKeyDown={false}
+      onClose={props.onClose}
+      open={true}
+    >
+      <DialogHeader mb={4}>
+        <DialogTitle>Connect To Database</DialogTitle>
+      </DialogHeader>
+
+      {attempt.status === 'error' && <Danger children={attempt.statusText} />}
+      {(attempt.status === '' || attempt.status === 'processing') && (
+        <Box textAlign="center" m={10}>
+          <Indicator />
+        </Box>
+      )}
+      {attempt.status === 'success' && (
+        <ConnectForm
+          db={attempt.data}
+          onConnect={props.onConnect}
+          onClose={props.onClose}
+        />
+      )}
+    </Dialog>
+  );
+}
+
+function ConnectForm(props: {
+  db: Database;
+  onConnect(data: DbConnectData): void;
+  onClose(): void;
+}) {
+  const dbUserOpts = props.db.users?.map(user => ({
+    value: user,
+    label: user,
+  }));
+  const dbNamesOpts = props.db.names?.map(name => ({
+    value: name,
+    label: name,
+  }));
+  const dbRolesOpts = props.db.roles?.map(role => ({
+    value: role,
+    label: role,
+  }));
+
+  const [selectedName, setSelectedName] = useState<Option>(dbNamesOpts?.[0]);
+  const [selectedUser, setSelectedUser] = useState<Option>(dbUserOpts?.[0]);
+  const [selectedRoles, setSelectedRoles] = useState<readonly Option[]>();
+
+  const dbConnect = () => {
+    props.onConnect({
+      serviceName: props.db.name,
+      protocol: props.db.protocol,
+      dbName: selectedName.value,
+      dbUser: selectedUser.value,
+      dbRoles: selectedRoles?.map(role => role.value),
+    });
+  };
+
+  return (
+    <Validation>
+      {({ validator }) => (
+        <form>
+          <DialogContent minHeight="240px" flex="0 0 auto">
+            <FieldSelectCreatable
+              label="Database user"
+              menuPosition="fixed"
+              onChange={option => setSelectedUser(option as Option)}
+              value={selectedUser}
+              options={dbUserOpts}
+              isDisabled={dbUserOpts?.length == 1}
+              formatCreateLabel={userInput =>
+                `Use "${userInput}" database user`
+              }
+              rule={requiredField('Database user is required')}
+            />
+            {dbRolesOpts?.length > 0 && (
+              <FieldSelect
+                label="Database roles"
+                menuPosition="fixed"
+                isMulti={true}
+                onChange={setSelectedRoles}
+                value={selectedRoles}
+                options={dbRolesOpts}
+              />
```

Comment:
About the wildcard for roles, we don't have enough information to decide when to show this box. This would be a combination of the database having auto-user provisioning enabled and not having DAC configured on their selected roles (correct me if I need to include anything here). So, for now, only showing when the roles are available might not cover all scenarios, but it will work for the most part. WDYT?

Also worth mentioning that currently, the database roles information is not available. The web handler changes will add this.

Reply:
just some facts
- database roles do not support wildcard
- database roles section is a very niche feature (like I mentioned, even the requestor of this feature might not be using it at the moment)
- database roles and DAC permissions are mutually exclusive

So for simplicity, we could just leave this part out.

If we want to keep it:
> So, for now, only showing when the roles are available might not cover all scenarios, but it will work for the most part.

this should be ok, also it's backend's responsibility to make sure the calculation is correct so we don't build more logic in frontend

> I think selecting all roles initially might be better here.

I think this would be easiest as well. 

> We can also always enforce at least one selected (when there are options available).

IMHO not necessary